Fix GNU stack

Quoted from https://src.fedoraproject.org/rpms/e3/blob/f30/f/e3.spec

# mark the stack as non-executable and disable tiny/crippled elf on 32
# bit linux so that stack can be marked as non-executable on it too

Downloaded from
https://src.fedoraproject.org/rpms/e3/blob/f30/f/e3-gnu-stack.patch

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

Common subdirectories: e3-2.8.orig/bin and e3-2.8/bin
diff -u e3-2.8.orig/e3.asm e3-2.8/e3.asm
--- e3-2.8.orig/e3.asm	2010-05-09 12:16:47.000000000 -0400
+++ e3-2.8/e3.asm	2013-02-22 14:24:46.884384131 -0500
@@ -6553,3 +6553,13 @@
 %ifdef CRIPLED_ELF
  bsssize equ $-bssstart
 %endif
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
Only in e3-2.8: e3.asm~
diff -u e3-2.8.orig/Makefile e3-2.8/Makefile
--- e3-2.8.orig/Makefile	2010-05-27 14:58:59.000000000 -0400
+++ e3-2.8/Makefile	2013-02-22 15:38:57.729605772 -0500
@@ -42,15 +42,10 @@
 #---------------- 32 bit LINUX (and perhaps BSD) --
 
 32:	$(ASOURCES) Makefile
-ifeq	($(OS),LINUX)
-	nasm -O2 -f bin -l e3.lst -o e3 e3.asm -DCRIPLED_ELF=1 -D$(OS)
-	chmod +x e3
-else
 	echo $(ASVER)
 	nasm -O2 $(AFLAGS) -o e3.o e3.asm -l e3.lst -D$(OS)
 	$(LD) -s -o e3 e3.o
 	strip --remove-section .comment e3
-endif
 	ln -sf e3 e3ws
 	ln -sf e3 e3em
 	ln -sf e3 e3pi
